kernel/src/checkpoint/log_replay.rs

```diff
@@ -0,0 +1,609 @@
+//! The [`CheckpointLogReplayProcessor`] implements specialized log replay logic for creating
+//! checkpoint files. It processes log files in reverse chronological order (newest to oldest)
+//! and selects only the minimal set of actions needed to represent the table state at a given version.
+//!
+//! ## Filtering Process
+//!
+//! For checkpoint creation, this processor applies several filtering and deduplication
+//! steps to each batch of log actions:
+//!
+//! 1. **Protocol and Metadata**: Retains only the latest protocol and metadata actions.
+//! 2. **Transactions**: Keeps the most recent action for each unique transaction (by app ID).
```

Comment:
can we reword to just be very specific that we keep one `txn` action for each app id and it's just the latest for that app id (and maybe instead of calling this `Transactions` let's use `SetTransactions` or `txn actions`